### DISCUSSION TITLE
Help with Simplified/Traditional Chinese translation via Weblate

### DISCUSSION THREAD

**Discussion by dangxit** [ncosgray/cuppa_mobile#104] [General]
Hi, I want to translate the app to Simplified/Traditional Chinese. However, when I tried to start a new translation on Weblate, it gives me "The given language is filtered by the language filter“ error. I wonder if Simplified/Traditional Chinese was disabled somehow? If so, would someone please enable it on Weblate so I can contribute to the translation.
Thank you for the app! I've been using it on Android and it's just wonderful to use. Hope you have a good day.

**Comment by ncosgray** [comment]
Oh, that would be great! I'd love it if Cuppa was translated to Chinese! Thank you!

I'm sorry Weblate isn't allowing you to start a translation for Simplified/Traditional Chinese. I tried it myself and saw the same error message. It's very strange -- there's no such thing as a "language filter" in the Weblate settings, so I'm not sure how to address this. I sent an email to the Weblate administrators asking what to do, and I'll post here when I get a response.

**Comment by ncosgray** [comment]
Actually, I think I have figured this out. Please try starting your translation, and it should work now! We may only be able to have one Chinese translation for all the Chinese language options (Simplified, Traditional, etc) so I hope that is alright.

Please let me know how it goes and if there are any more problems.

**Reply by dangxit** [reply]
Awesome 👍 I will start on the translation asap, and comment if there's any issue. Thank you!

**Comment by dangxit** [comment]
Hi there. I have translated Cuppa mobile app and macOS app into simplified Chinese. I don't personally own a macOS so there might be context errors with the translation (just a heads up).
Another question with Weblate: When I tried to add Traditional Chinese on weblate, the website gives me "The given Language already exists" error. I'm not sure if weblate automatically replaces Simplified Chinese characters to Traditional Chinese characters (since grammar for the two languages are the same), but if it doesn't you might want to look into it since some of the characters are different. 
Thanks and hope you have a good one!

**Comment by ncosgray** [comment]
Thank you so much for contributing the Chinese translation for Cuppa! This is a really exciting addition!

I just added an onboarding tutorial for new users that explains how the app works. There are a handful of new strings to translate. Apologies. :)

Also, unfortunately, there is a limitation in the Cuppa code right now which means only one variant of a particular language is supported. So we can only have one language code "zh" (Chinese) translation for the app, and anyone with a phone set to any Chinese language -- whether it's Simplified, Traditional, etc. -- will see that translation. It'll take some effort to work things out in the app code to allow other variants like Traditional Chinese to be added to the app. I'll post here when I've been able to do that.

**Comment by dangxit** [comment]
Ahh I see. No worries, I'm always happy to contribute to the project 😄. As for traditional Chinese, most people who know one variation can understand the other so no rush. Thank you for taking it into consideration! Hope you have a good one

**Comment by ncosgray** [comment]
@dangxit, I've been able to implement support for language scripts in Cuppa, so we can have separate translations for Traditional and Simplified Chinese now!

I went ahead and added an auto-translated set of Traditional Chinese strings to Weblate: https://hosted.weblate.org/projects/cuppa/cuppa-mobile-app/zh_Hant/

No doubt the auto-translated results need some tweaking. Please take a look whenever you have some free time. Thanks!

**Reply by dangxit** [reply]
Awesome! Thank you for implementing multi language support for traditional Chinese! I will take a look at traditional Chinese string when I have free time.
